xpasses/XS002/XS002.go

```diff
@@ -0,0 +1,55 @@
+package XS002
+
+import (
+	"go/ast"
+	"sort"
+
+	"github.com/bflad/tfproviderlint/helper/astutils"
+	"github.com/bflad/tfproviderlint/helper/terraformtype/helper/schema"
+	"github.com/bflad/tfproviderlint/passes/commentignore"
+	"github.com/bflad/tfproviderlint/passes/helper/schema/schemamapcompositelit"
+	"golang.org/x/tools/go/analysis"
+)
+
+const Doc = `check for Schema that attribute names are in alphabetical order
+
+The XS002 analyzer reports cases of schemas where attribute names
+are not in alphabetical order.`
+
+const analyzerName = "XS002"
+
+var Analyzer = &analysis.Analyzer{
+	Name: analyzerName,
+	Doc:  Doc,
+	Requires: []*analysis.Analyzer{
+		schemamapcompositelit.Analyzer,
+		commentignore.Analyzer,
+	},
+	Run: run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	ignorer := pass.ResultOf[commentignore.Analyzer].(*commentignore.Ignorer)
+	schemamapcompositelits := pass.ResultOf[schemamapcompositelit.Analyzer].([]*ast.CompositeLit)
+
+	for _, smap := range schemamapcompositelits {
+		if ignorer.ShouldIgnore(analyzerName, smap) {
+			continue
+		}
+
+		schemaKeys := make([]string, 0 , len(schema.GetSchemaMapAttributeNames(smap)))
+		for _, attributeName := range schema.GetSchemaMapAttributeNames(smap) {
+			if v := astutils.ExprStringValue(attributeName); v != nil {
+				schemaKeys = append(schemaKeys, *v)
+			} else if v := astutils.IdentExprStringValue(attributeName); v != nil {
+				schemaKeys = append(schemaKeys, *v)
+			}
```

Comment:
I'm not sure we want to account for any constant/variable naming in the ordering, since their naming can be arbitrary compared to the rest of the keys. Take this example:

```go
Schema: map[string]*schema.Schema{
  regionSchemaKey: regionSchema(),
  "a_attr": {},
  "z_attr": {},
},
```

Which may not be desirable to enforce this:

```go
Schema: map[string]*schema.Schema{
  "a_attr": {},
  regionSchemaKey: regionSchema(),
  "z_attr": {},
},
```

I think its best we just stick to literal strings and allow downstream tooling to enforce more specific rules if they wish. 👍 

```suggestion
			}
```

Reply:
sounds good!